Sources/Apodini/Properties/ObservedObject.swift

```diff
@@ -0,0 +1,101 @@
+import Foundation
+
+/// Property wrapper that subscribes to an `ObservableObject` and evaluates a `Handler` or `Job` on changes.
+///
+/// Refer to the documentation of
+/// [ObservedObject](https://github.com/Apodini/Apodini/blob/develop/Documentation/Communicational%20Patterns/2.%20Tooling/2.4.%20ObservedObject.md)
+/// for more information.
+@propertyWrapper
+public struct ObservedObject<Element: ObservableObject>: Property {
+    private var objectIdentifer: ObjectIdentifier?
+    private var element: Element?
+    private var changedWrapper: Wrapper<Bool>
+    private var wrappedValueDidChange: Wrapper<(() -> Void)?>
+    
+    public var wrappedValue: Element {
+        get {
+            if let element = element {
+                return element
+            }
+            if let objectIdentifer = objectIdentifer,
+               let element = EnvironmentValues.shared.values[objectIdentifer] as? Element {
+                return element
+            }
+            fatalError("The object \(String(describing: self)) cannot be found in the environment.")
+        }
+        set {
+            element = newValue
+        }
+    }
+    
+    /// Property to check if the evaluation of the `Handler` or `Job` was triggered by this `ObservableObject`.
+    /// Read only property
+    public var changed: Bool {
```

Comment:
Just a suggestion...

```suggestion
public enum ObservedObjectState: Bool {
    case changed = true
    case unchanged = false
}

    public var projectedValue: ObservedObjectState {
```
One could then write just `$someObservedObject` instead of `_someObservedObject.changed` with the returned type still making clear what the value means.

I'm not sure if you could copy this code 1:1, but I think you get the gist. 

This is just some syntactic sugar and it totally depends on taste, so you can also just leave it as is :).

Reply:
I like the use `projectedValue` and think that it is more swift-like than using `_`. However, I think when just using `$someObservedObject` the return type might be a bit confusing because it is not directly object related and returns something else. For this reason I think `_someObservedObject.changed` better serves the purpose.